proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+| **Architecture**       | Separate object storage service                 | KFP-native artifact server                        |
+| **Required Knowledge** | S3 concepts (buckets, endpoints, regions)       | Kubernetes concepts (PVCs, StorageClasses)        |
+| **Multi-tenancy**      | Shared storage (single instance)                | Per-namespace PVCs (in namespace-local mode)      |
+| **Scaling**            | Single object storage instance                  | Per-namespace artifact servers and storage quotas |
+
+### High-Level Flow
+
+```text
+Pipeline Submission
+        │
+        ▼
+Driver (detects "kfp-artifacts://")
+        │
+        ▼
+Artifact Server (mounts PVC)
+        │
+        ├──────────────────────────────────┐
+        │                                  │
+        ▼                                  ▼
+Pipeline Pods                         UI/API Clients
+(upload/download via API)             (view artifacts)
+        │                                  │
+        └──────────► Artifact Server ◄─────┘
+                           │
+                           ▼
+                    Read/Write to PVC
+```
+
+### Key Components
+
+- **Storage**: Kubernetes `PersistentVolumeClaims` (one per namespace)
+- **URI Format**: `kfp-artifacts://<namespace>/<pipeline>/<run-id>/<node-id>/<artifact-name>`
+- **API Changes**: New `/filesystem-storage/config` endpoint for UI configuration discovery
+- **Deployment Modes**:
+  - **Central**: Single artifact server for all namespaces (default)
+  - **Namespace-local**: One artifact server per namespace (better isolation)
+  - **Mixed**: Per-namespace mode override via `kfp-launcher` ConfigMap
+
+## Motivation
+
+**This proposal does not aim to replace existing object storage solutions.** S3-compatible storage remains fully supported and recommended for most production workloads. Instead, this KEP provides an additional option for deployments where a simpler, KFP-native artifact storage solution is preferred.
+
+While KFP currently ships with S3-compatible storage by default, this still requires deploying and maintaining a separate object storage service. For some deployment scenarios, this additional component may not be desired.
+
+### Reduced External Dependencies
+
+While KFP's default manifests ship with S3-compatible storage pre-configured (no manual credential setup required), filesystem storage replaces the object storage system with a KFP-native artifact server. This means:
+
+- No external object storage project to track for updates and security patches
+- Artifact issues debugged within KFP ecosystem (no separate S3 API layer)
+- Artifact handling uses the same codebase as the rest of KFP
+- No S3-specific knowledge required (buckets, endpoints, regions). Only familiar Kubernetes concepts (PVCs, `StorageClasses`)
+
+### Enterprise Considerations
+
+Many enterprises and Kubeflow distributions prefer not to have additional external dependencies. With filesystem storage:
```

Comment:
Another aspect is you'll automatically get namespace isolation of artifacts on the central artifact server through namespace aware paths and Kubernetes RBAC.